server/src/logic/services/auth.service.ts

```diff
@@ -0,0 +1,96 @@
+import * as jwt from "jsonwebtoken";
+import { UserService } from "./user.service";
+import { CryptoUtils } from "./../utils/cryptoUtils";
+import { ErrorUtils } from "./../utils/errorUtils";
+import { Config } from "./../../config";
+import { ApiError, DbError, SocoboUser } from "./../../models/index";
+
+
+export class AuthService {
+  constructor (private _userService: UserService) {}
+
+  login (email: string, password: string): Promise<Object> {
+    return new Promise((resolve, reject) => {
+      // search for the user by provided email
```

Comment:
Yes you are right. I add comments because I hate whitespace. But these functions do too much. That's a good idea to break things down into smaller functions. I really need to read Clean code. ;-)